File: Array/325_Maximum Size Subarray Sum Equals k.py
class Solution:
    def maxSubArrayLen(self, nums: List[int], k: int) -> int:
        res = 0
        total = 0
        ht = {0: -1}
        for i, num in enumerate(nums):
            total += num
            if total - k in ht:
                res = max(res, i - ht[total - k])
            if total not in ht:
                ht[total] = i
        return res

File: Java Solutions/325_MaximumSizeSubarraySumEqualsK.java
class Solution {
    public int maxSubArrayLen(int[] nums, int k) {
        int curSum = 0, res = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0, -1);
        for (int i = 0; i < nums.length; i++) {
            curSum += nums[i];
            if (map.containsKey(curSum - k)) {
                res = Math.max(res, i - map.get(curSum - k));
            }
            if (!map.containsKey(curSum)) {
                map.put(curSum, i);
            }
        }
        return res;
    }
}